server/controllers/message.js

```diff
@@ -0,0 +1,92 @@
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+const Conversation = require("../models/Conversation");
+const Message = require("../models/Message");
+
+const { findConversationQuery } = require("../utils/conversationQueries");
+
+// @route GET /messages/:convoId
+// @desc retrieves all messages by a given conversation id
+// @access Private
+exports.getAllMessages = asyncHandler(async (req, res) => {
+  const {
+    params: { convoId },
+    query: { sort = "createdAt", order = "asc" },
+  } = req;
+  const limit = parseInt(req.query.limit || 30),
+    page = parseInt(req.query.page || 1);
+
+  const messages = await Message.find({
+    conversationId: convoId,
+  })
+    .limit(limit)
+    .lean()
+    .skip((page - 1) * limit)
+    .populate({
+      path: "sender",
+      select: {
+        _id: 1,
+        name: 1,
+        photo: 1,
+      },
+    })
+    .sort({
+      [sort]: order,
+    })
+    .exec();
+
+  return res.status(200).json({
+    success: {
+      messages: messages.map(({ _id, ...message }) => {
+        message.id = _id;
+        return message;
+      }),
+    },
+  });
+});
+
+// @route POST /messages/send
+// @desc Creates a new message.
+// @access Private
+exports.sendMessage = asyncHandler(async (req, res) => {
+  const {
+    user,
+    body: { receiverId, content },
+  } = req;
```

Comment:
Maybe it makes more sense to recieve conversation id here. This would work better if/when we scale them to have groups of users too.

Reply:
While I agree that it would make sense to receive a conversation id in order to send a message and I can certainly make those changes. If I were to implement it to include groups, I would need to redesign the conversation model to instead store an array of referenced profiles. I agree it would work in terms of scalability in case we do implement group messaging, but my understanding of the project and the information that was provided, it does not look like we need to scale into group messaging. Let me know if you want me to make the changes though, in order to allow for groups @ugur-akin  :+1:

Reply:
You do have a good point, maybe we should hold off on that 👍🏼